### PR TITLE
Update Docker docs for v6

### PR DESCRIPTION
**Pull request by jarpy**
* Document Docker image "flavours"
* Fix up section hierarchy
* Capture Docker config default
* Warn about default "masking" when replacing logstash.yml

### CODE REVIEW

File: docs/static/docker.asciidoc [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
+found on https://github.com/elastic/logstash-docker/tree/{branch}[GitHub].
 
-The images are shipped with https://www.elastic.co/products/x-pack[X-Pack]
-installed.
+==== Image types
+
+The images are available in two different configurations or "flavors". The
+`x-pack` flavor, which is the default, ships with X-Pack features
+pre-installed. The `oss` flavor does not include X-Pack, and contains only
+open-source Logstash.
```
Comment (michaelbaamonde) [inline docs/static/docker.asciidoc]: Super nitpicky, but the internal style guide says not to hyphenate "open source."

Comment (jarpy) [reply docs/static/docker.asciidoc]: Thanks. Super nitpicky is what we do!

**Comment by jarpy** [comment]
What is the correct way to get this docs patch into `6.x`/`6.0`?

`git cherry-pick` or something more sophisticated?

**Comment by lcawl** [comment]
There was a broken link in this update, so I've put a fix in commit 9ef86d0d67e73d9966ae8cb0d60978e9893363b1

**Comment by jarpy** [comment]
`6.0` -> 712acbf
`6.x` -> a7bb930

**Pull request merged**